core/variables.js

```diff
@@ -173,13 +173,19 @@ Blockly.Variables.generateUniqueName = function(workspace) {
  * @param {string} opt_type Optional type of variable, like 'string' or 'list'.
  */
 Blockly.Variables.createVariable = function(workspace, opt_callback, opt_type) {
-  // (karishma) TODO (#1244) Modal message should change depending on what type
-  // (opt_type above) of variable we are creating. If opt_type is not provided,
-  // we should default to the original 'NEW_VARIABLE_TITLE' message for
-  // scalar variables.
+  // Decide on a modal message based on the opt_type. If opt_type was not
+  // provided, default to the original message for scalar variables.
+  var newMsg = '';
+  if (opt_type === 'list') {
+    newMsg = Blockly.Msg.NEW_LIST_TITLE;
```

Comment:
'list' should be a constant, and 'broadcast_msg' already is.

Reply:
done.